evm/src/trace/mod.rs

```diff
@@ -60,20 +63,23 @@ impl CallTraceArena {
         }
     }
 
-    pub fn addresses_iter(&self) -> impl Iterator<Item = (&Address, Option<&Vec<u8>>)> {
-        self.arena.iter().map(|node| {
-            let code = if node.trace.created() {
-                if let RawOrDecodedReturnData::Raw(bytes) = &node.trace.output {
-                    Some(bytes)
+    pub fn addresses(&self) -> HashSet<(&Address, Option<&Vec<u8>>)> {
+        self.arena
+            .iter()
+            .map(|node| {
+                let code = if node.trace.created() {
+                    if let RawOrDecodedReturnData::Raw(bytes) = &node.trace.output {
+                        Some(bytes)
```

Comment:
can return `(&node.trace.address, Some(..))` here
and default is (&node.trace.address, None)